### PR TITLE
Convert yargs builder functions to options objects and add per–command environment variables

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Fixes https://github.com/iterative/cml/issues/708 along with some minor creative bugs we've been overlooking for months. 🐞

**Contains cascaded https://github.com/iterative/cml/pull/739**

### CODE REVIEW

File: bin/cml/runner.js
Comment (0x2b3bfa0) [inline bin/cml/runner.js:538]:
```suggestion
  tfResource: {
    hidden: true,
    alias: 'tf_resource'
  },
```

File: bin/cml/runner.test.js
Comment (0x2b3bfa0) [inline bin/cml/runner.test.js:66]:
```suggestion
                                                          [string] [default: \\"\\"]"
```

**Review by DavidGOrtega — COMMENTED** [review]
Looking good.

I think that we can make it DRY with repo, token and driver

**Comment by 0x2b3bfa0** [comment]
@DavidGOrtega, is your review comment an implicit approval?

**Comment by DavidGOrtega** [comment]
No

**Comment by 0x2b3bfa0** [comment]
> _However maybe thats not a stopper. I have not approved sometimes I we have still merged_ 🙂

It is. 🙃  I don't recall merging any pull request with zero reviews, apart from those old release branches that were intended to bump the package version.

> _I would consider to have all those shared params in one place maybe merging with destructure assignment_

As far as I know, all the CML commands – except for `tensorboard-dev` – require `--repo` and `--token` to work, so it would make sense to group them somehow. Nevertheless, this kind of change doesn't seem to have anything to do with the title of this pull request. Maybe it would be better to leave it for a follow–up (?)

**Comment by DavidGOrtega** [comment]
> It is. 🙃 I don't recall merging any pull request with zero reviews, apart from those old release branches that were intended to bump the package version.

Not with zero reviews but with opposed reviews 🙂

**Comment by DavidGOrtega** [comment]
> As far as I know, all the CML commands – except for tensorboard-dev – require --repo and --token to work, so it would make sense to group them somehow. Nevertheless, this kind of change doesn't seem to have anything to do with the title of this pull request. Maybe it would be better to leave it for a follow–up (?)

Also driver. I found interesting the opportunity to make it here. Not difficult either. Whatever.

**Comment by 0x2b3bfa0** [comment]
> _Not with zero reviews but with opposed reviews_ <sup><b>[citation needed]</b></sup> 🙂

(Off–topic) Sorry if I did. I don't recall any particular case and, usually, wait for approvals to be green before merging.

**Comment by 0x2b3bfa0** [comment]
> _Also driver. I found interesting the opportunity to make it here. Not difficult either. Whatever._

Definitely not difficult, but rather the opposite: top–level options get automatically propagated without having to write a signle line of code. Nevertheless, I'd prefer to keep #712 on scope and perform this change separately.

We should probably find out what to do with `tensorboard-dev`

**Comment by 0x2b3bfa0** [comment]
@DavidGOrtega, then, can we merge this as–is and delay common option unification?

**Review by DavidGOrtega — APPROVED** [review]

**Pull request merged**
